File: SPCode/CS/Modules/Encrypt.cs
using System.Text.RegularExpressions;
using System.Diagnostics;
using System;
using System.Drawing.Drawing2D;
using System.Windows.Forms;
using System.Collections;
using System.Drawing;
using Microsoft.VisualBasic;
using System.Data.SqlClient;
using System.Data;
using System.Collections.Generic;
using WeifenLuo.WinFormsUI;
using Microsoft.Win32;
using WeifenLuo;




namespace SoftLogik.Win
{
	public enum HashTypeEnum
	{
		htMD5 = 4
	}
	
	public enum EncryptAlgorithmsEnum
	{
		eaRC4 = 1,
		eaRC2,
		eaDES,
		ea3DES,
		ea3DES112
	}
	
	public sealed class SPEncrypt
	{
		
		public static byte[] EncryptByte(EncryptAlgorithmsEnum Algorithm, string SourceText, bool CaseSensitive, string Key, string Salt)
		{
			CryptKci.clsCryptoAPI objCrypto = new CryptKci.clsCryptoAPI();
			
			try
			{
				objCrypto.InputData = objCrypto.StringToByteArray((object) SourceText);
				objCrypto.EnhancedProvider = true;
				objCrypto.Password = objCrypto.StringToByteArray((object) Key);
				objCrypto.Encrypt(((short) HashTypeEnum.htMD5), ((short) Algorithm));
				return objCrypto.OutputData;
				
			}
			catch (Exception ex)
			{
				throw (ex);
			}
			finally
			{
				objCrypto = null;
			}
			
		}
		
		public static string EncryptString(EncryptAlgorithmsEnum Algorithm, string SourceText, bool CaseSensitive, string Key, string Salt)
		{
			CryptKci.clsCryptoAPI objCrypto = new CryptKci.clsCryptoAPI();
			
			try
			{
				objCrypto.InputData = objCrypto.StringToByteArray(System.Convert.ToString(SourceText));
				objCrypto.EnhancedProvider = true;
				objCrypto.Password = objCrypto.StringToByteArray(System.Convert.ToString(Key));
				objCrypto.Encrypt(((short) HashTypeEnum.htMD5), ((short) Algorithm));
				return objCrypto.ByteArrayToString(objCrypto.OutputData);
				
			}
			catch (Exception ex)
			{
				throw (ex);
			}
			finally
			{
				objCrypto = null;
			}
			
		}
		public static byte[] ToByte(string SourceString)
		{
			CryptKci.clsCryptoAPI objCrypto = new CryptKci.clsCryptoAPI();
			
			return objCrypto.StringToByteArray(System.Convert.ToString(SourceString));
//			objCrypto = null;
			
		}
		
		public static string CreateSaltValue(int ReturnLength, bool UseLettersNumbersOnly)
		{
			CryptKci.clsCryptoAPI objCrypto = new CryptKci.clsCryptoAPI();
			
			return System.Convert.ToString(objCrypto.CreateSaltValue(ref ReturnLength, ref UseLettersNumbersOnly));
//			objCrypto = null;
			
		}
	}
	
	
}

File: SPCode/Modules/Encrypt.vb



Public Enum HashTypeEnum
    htMD5 = 4
End Enum

Public Enum EncryptAlgorithmsEnum
    eaRC4 = 1
    eaRC2
    eaDES
    ea3DES
    ea3DES112
End Enum

Public Module SPEncrypt

    Public Function EncryptByte(ByVal Algorithm As EncryptAlgorithmsEnum, ByVal SourceText As String, ByVal CaseSensitive As Boolean, ByVal Key As String, ByVal Salt As String) As Byte()
        Dim objCrypto As New CryptKci.clsCryptoAPI

        Try
            With objCrypto
                .InputData = .StringToByteArray(CType(SourceText, Object))
                .EnhancedProvider = True
                .Password = .StringToByteArray(CType(Key, Object))
                .Encrypt(CType(HashTypeEnum.htMD5, Short), CType(Algorithm, Short))
                Return .OutputData
            End With

        Catch ex As Exception
            Throw ex
        Finally
            objCrypto = Nothing
        End Try

    End Function

    Public Function EncryptString(ByVal Algorithm As EncryptAlgorithmsEnum, ByVal SourceText As String, ByVal CaseSensitive As Boolean, ByVal Key As String, ByVal Salt As String) As String
        Dim objCrypto As New CryptKci.clsCryptoAPI

        Try
            With objCrypto
                .InputData = .StringToByteArray(CType(SourceText, String))
                .EnhancedProvider = True
                .Password = .StringToByteArray(CType(Key, String))
                .Encrypt(CType(HashTypeEnum.htMD5, Short), CType(Algorithm, Short))
                Return .ByteArrayToString(.OutputData)
            End With

        Catch ex As Exception
            Throw ex
        Finally
            objCrypto = Nothing
        End Try

    End Function
    Public Function ToByte(ByVal SourceString As String) As Byte()
        Dim objCrypto As New CryptKci.clsCryptoAPI

        Return objCrypto.StringToByteArray(CType(SourceString, String))
        objCrypto = Nothing

    End Function

    Public Function CreateSaltValue(Optional ByVal ReturnLength As Integer = 20, _
    Optional ByVal UseLettersNumbersOnly As Boolean = True) As String
        Dim objCrypto As New CryptKci.clsCryptoAPI

        Return CType(objCrypto.CreateSaltValue(ReturnLength, UseLettersNumbersOnly), String)
        objCrypto = Nothing

    End Function
End Module

